### PR TITLE
[DOC] RDoc for Process::Status

### PR DESCRIPTION
**Pull request by BurdetteLamar**

### CODE REVIEW

File: process.c [hunk 1]
```diff
@@ -566,30 +566,22 @@ proc_get_ppid(VALUE _)
...
- *  exited, signaled) and the upper bits possibly contain additional
- *  information (for example the program's return code in the case of
- *  exited processes). Pre Ruby 1.8, these bits were exposed directly
- *  to the Ruby program. Ruby now encapsulates these in a
- *  Process::Status object. To maximize compatibility,
- *  however, these objects retain a bit-oriented interface. In the
- *  descriptions that follow, when we talk about the integer value of
- *  _stat_, we're referring to this 16 bit value.
+ *  A Process::Status may contain (and usually does contain)
+ *  information about a system process.
```
Comment (peterzhu2118) [inline process.c:570]: I'm not sure when `Process::Status` doesn't contain information about a system process?

File: process.c [hunk 2]
```diff
@@ -716,14 +708,12 @@ pst_status(VALUE pst)
...
  *
- *  Returns the bits in _stat_ as an Integer. Poking
- *  around in these bits is platform dependent.
+ *  Returns the system-dependent integer status of +self+:
  *
- *     fork { exit 0xab }         #=> 26566
- *     Process.wait               #=> 26566
- *     sprintf('%04x', $?.to_i)   #=> "ab00"
+ *    `cat /nop`
+ *    $?.to_i # => 256*
```
Comment (peterzhu2118) [inline process.c:716]: Is the asterisk here a typo?

File: process.c [hunk 3]
```diff
@@ -826,12 +817,12 @@ pst_to_s(VALUE st)
...
  *  call-seq:
- *     stat.inspect   -> string
+ *    inspect -> string
  *
- *  Override the inspection method.
+ *  Returns a string representation of +self+:
  *
  *    system("false")
- *    p $?.inspect #=> "#<Process::Status: pid 12861 exit 1>"
+ *    $? # => #<Process::Status: pid 1246759 exit 1>
```
Comment (peterzhu2118) [inline process.c:825]: We should explicitly call `inspect` because it's only IRB that implicitly calls `inspect` for the output.

File: process.c [hunk 4]
```diff
@@ -942,10 +933,10 @@ pst_wifstopped(VALUE st)
 
 /*
  *  call-seq:
- *     stat.stopsig   -> integer or nil
+ *    stopsig -> integer or nil
  *
- *  Returns the number of the signal that caused _stat_ to stop
- *  (or +nil+ if self is not stopped).
+ *  Returns the number of the signal that caused the process to stop,
+ *  or +nil+ if the process is not stopped,
```
Comment (peterzhu2118) [inline process.c:939]:
```suggestion
 *  or +nil+ if the process is not stopped.
```

File: process.c (hunk 1 above)
Comment (BurdetteLamar) [reply process.c:570]: Line 1174 has what I thought of as an "empty" Process::Status object;  not so?

File: process.c (hunk 2 above)
Comment (BurdetteLamar) [reply process.c]: Yup;  fixed.

File: process.c (hunk 3 above)
Comment (BurdetteLamar) [reply process.c]: Duh.  inspect is the method being docked.  Fixed.

File: process.c (hunk 1 above)
Comment (peterzhu2118) [reply process.c:570]: I see. While you're technically correct, I think that saying "may contain" is confusing to the reader and it leaves then guessing in what cases this doesn't contain valid information.

Comment (BurdetteLamar) [reply process.c]: Fixed.

**Review by peterzhu2118 — APPROVED** [review]

File: process.c (hunk 1 above)
Comment (peterzhu2118) [reply process.c]: Thank you!

**Pull request merged**